mitodl/open-discussions
# Fix some attributes and state for KC export

#### What are the relevant tickets?
N/A

#### What's this PR do?

- Switches `createdTimestamp` to an integer in milliseconds. The previous stringified value was being parse typically as some datetime in January of 1970. No clear reason why, no documentation as to what it should be aside from a `long`. API didn't seem to reject the value though, although arbitrary string values _did_ get rejected. Timestamps in seconds had the same result.
- Added a `--verbose` flag that logs the requests, error responses are still always logged
- Added handling for skipped users, with a new `action` field on the model to reflect whether a user was actually exported or skipped because they already exist. Without this users who already existed were attempted to export perpetually.

#### How should this be manually tested?
Run `./manage.py migrate_users_to_keycloak --client-id admin-cli --client-secret SECRET --verbose` and you should see verbose output.
